File: 백준/Silver/9012. 괄호/괄호.py
testNun = int(input())

for _ in range(testNun):
    inputData = input()
    count = 0

    for i in inputData:
        if i == '(':
            count += 1
        elif i == ')':
            count -= 1
        if count < 0:
            print('NO')
            break
            
    if count > 0:
        print("NO")
    elif count == 0:
        print('YES')

File: 백준/Silver/9012. 괄호/괄호.swift
import Foundation

func check(_ s: String) -> Bool {
    
    var stk: [Character] = []
    
    for c in s {
        if c == "(" {
            stk.append(c)
        } else if c == ")" {
            if !stk.isEmpty {
                stk.removeLast()
            } else {
                return false
            }
        }
    }
    
    return stk.isEmpty
}

var n = Int(readLine()!)!

for _ in 0..<n {
    let input = readLine()!
    if check(input) {
        print("YES")
    } else {
        print("NO")
    }
}
